src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -528,7 +528,7 @@ private IEnumerable<LibraryDependency> GetLibraryDependenciesForCentralTransitiv
                     CentralPackageVersion centralPackageVersion = targetFrameworkInformation.CentralPackageVersions[node.Item.Key.Name];
                     Dictionary<string, LibraryIncludeFlags> dependenciesIncludeFlags = _includeFlagGraphs[targetGraph];
 
-                    LibraryIncludeFlags suppressParent = LibraryIncludeFlags.None;
+                    LibraryIncludeFlags suppressParent = LibraryIncludeFlags.All;
```

Comment:
Should this be conditional depending on if `centralPackageTransitivePinningEnabled` is true or false?
Now it's always `LibraryIncludeFlags.All`.

Reply:
I've pushed a change that checks whether the `centralPackageTransitivePinningEnabled` is true before calling the `GetLibraryDependenciesForCentralTransitiveDependencies`.